tests/unit/models/ObjectMD.js

```diff
@@ -257,6 +257,23 @@ describe('ObjectMD class setters/getters', () => {
         md.clearMetadataValues();
         assert.strictEqual(md.getUserMetadata(), undefined);
     });
+
+    it('ObjectMD::microVersionId', () => {
+        assert.strictEqual(md.getMicroVersionId(), null);
+
+        const generatedIds = {};
+        for (let i = 0; i < 100; ++i) {
+            md.updateMicroVersionId();
+            generatedIds[md.getMicroVersionId()] = 1;
+        }
+        // all generated IDs should be different
+        assert.strictEqual(Object.keys(generatedIds).length, 100);
+        Object.keys(generatedIds).forEach(key => {
+            // length is always 16 in hex because leading 0s are
+            // also encoded in the 8-byte random buffer.
+            assert.strictEqual(key.length, 16);
```

Comment:
Instead of comparing lengths, you could mock the crypto call to ensure it is calling it with specific params

Reply:
see answer below